src/Microsoft.VisualStudio.Threading.Analyzers.CSharp/VSTHRD110ObserveResultOfAsyncCallsAnalyzer.cs

```diff
@@ -46,7 +46,7 @@ internal void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
 
                 // Only consider invocations that are direct statements. Otherwise, we assume their
                 // result is awaited, assigned, or otherwise consumed.
-                if (invocation.Parent?.GetType().Equals(typeof(ExpressionStatementSyntax)) ?? false)
+                if (invocation.Parent is ExpressionStatementSyntax || invocation.Parent is ConditionalAccessExpressionSyntax)
```

Comment:
tip: this can be written even more succinctly as:

```cs
if (invocation.Parent is ExpressionStatementSyntax or ConditionalAccessExpressionSyntax)
```

Reply:
Thanks! Good to know.